### DISCUSSION TITLE
Find DID

### DISCUSSION THREAD

**Discussion by RobertoSnap** [decentralized-identity/veramo#567] [Q&A] [answered]
Trying to find a DID: "did:key:z6Mkk1V84BS2VGjRyHEsC72w1FBpjLTtUN2bdqiHzpuxr9X8"

This is my current identifier table
provider | alias | saveDate | updateDate | controllerKeyId
-- | -- | -- | -- | --
did:key:z6Mkk1V84BS2VGjRyHEsC72w1FBpjLTtUN2bdqiHzpuxr9X8 | did:key | NULL | 2021-06-15 10:26:26 | 2021-06-15 10:26:26 | 528bce840023b038cf9d65fe7dcbe36b0a83302c84477b8b0e758cbe70eadacb
did:key:z6MkqPbRao8FPXY2i6xoLkpY2wJXa4aEMcjmXmNeWiBWEzYH | did:key | NULL | 2021-06-15 10:28:09 | 2021-06-15 10:28:09 | a2802522898f7792fee67f234d334cb2c0c1c3fa840dda361367dcbf9149b2f2
did:key:z6Mkge3wTYspATmbggVajdYXtHTWfwC3icYDorHFZMpGeeyj | did:key | NULL | 2021-06-15 10:29:25 | 2021-06-15 10:29:25 | 207b132305858b82973b1b4a6d3522d1364cdd4e0cef11b3545981277eb1c626

```
await this.agent.didManagerFind({
      provider: 'did:key:z6Mkk1V84BS2VGjRyHEsC72w1FBpjLTtUN2bdqiHzpuxr9X8',
    });
```
But getting empty result back. What am I doing wrong?

https://github.com/BROKLab/auth-bot/blob/veramo/src/network/veramo.service.ts#L87

**Comment by mirceanis** [comment, marked as answer]
try `didManagerGet` instead of `didManagerFind`

```ts
const identifier = await this.agent.didManagerGet({
      did: 'did:key:z6Mkk1V84BS2VGjRyHEsC72w1FBpjLTtUN2bdqiHzpuxr9X8',
    });
```

**Reply by RobertoSnap** [reply]
Works thanks!
